iml-api/src/graphql.rs

```diff
@@ -148,96 +153,251 @@ impl QueryRoot {
+        fsname(description = "Filesystem snapshot was taken from"),
+        name(description = "Name of the snapshot"),
+    ))]
+    async fn mount_snapshot(
+        context: &Context,
+        fsname: String,
+        name: String,
+    ) -> juniper::FieldResult<Command> {
+        let active_mgs_host_fqdn = active_mgs_host_fqdn(&fsname, &context.pg_pool).await?;
+
+        let kwargs: HashMap<String, String> = vec![("message".into(), "Mounting snapshot".into())]
+            .into_iter()
+            .collect();
+
+        let jobs = serde_json::json!([{
+            "class_name": "MountSnapshotJob",
+            "args": {
+                "fsname": fsname,
+                "name": name,
+                "fqdn": active_mgs_host_fqdn,
+            }
+        }]);
+        let command_id: i32 = iml_job_scheduler_rpc::call(
+            &context.rabbit_pool.get().await?,
+            "run_jobs",
+            vec![jobs],
+            Some(kwargs),
+        )
+        .map_err(ImlApiError::ImlJobSchedulerRpcError)
+        .await?;
+
+        get_command(&context.pg_pool, command_id)
+            .await
+            .map_err(|e| e.into())
+    }
+    #[graphql(arguments(
+        fsname(description = "Filesystem snapshot was taken from"),
+        name(description = "Name of the snapshot"),
+    ))]
+    async fn unmount_snapshot(
+        context: &Context,
+        fsname: String,
+        name: String,
+    ) -> juniper::FieldResult<Command> {
+        let active_mgs_host_fqdn = active_mgs_host_fqdn(&fsname, &context.pg_pool).await?;
+
+        let kwargs: HashMap<String, String> =
+            vec![("message".into(), "Unmounting snapshot".into())]
                 .into_iter()
-                .map(|x| Snapshot {
-                    snapshot_name: x.snapshot_name,
-                    filesystem_name: x.filesystem_name,
-                    details: vec![],
-                })
                 .collect();
 
-            Ok(snapshots)
-        }
+        let jobs = serde_json::json!([{
+            "class_name": "UnmountSnapshotJob",
+            "args": {
+                "fsname": fsname,
+                "name": name,
+                "fqdn": active_mgs_host_fqdn,
+            }
+        }]);
+        let command_id: i32 = iml_job_scheduler_rpc::call(
+            &context.rabbit_pool.get().await?,
+            "run_jobs",
+            vec![jobs],
+            Some(kwargs),
+        )
+        .map_err(ImlApiError::ImlJobSchedulerRpcError)
+        .await?;
+
+        get_command(&context.pg_pool, command_id)
+            .await
+            .map_err(|e| e.into())
     }
 }
 
+async fn active_mgs_host_fqdn(
+    fsname: &str,
+    pool: &PgPool,
+) -> Result<String, iml_postgres::sqlx::Error> {
+    let fsnames = &[fsname.into()][..];
+    let active_mgs_host_id = sqlx::query!(
+        r#"
+            select active_host_id from targets where filesystems @> $1 and name='MGS'
+            "#,
+        fsnames
+    )
+    .fetch_one(pool)
```

Comment:
This assumes that there will both be a fs that matches and that the MGS is mounted on a host. 

I suggest using `.fetch_optional(pool)` instead and having this fn return `Result<Option<String>, iml_postgres::sqlx::Error>`.

Callers will need to deal with the case where the MGS is unavailable; we should return an error describing this to users.

Reply:
Looks like this now
![image](https://user-images.githubusercontent.com/393741/92365814-033ddf00-f0fd-11ea-94bf-2a1594872764.png)